### PR TITLE
78595 - dev unitofsale event endpoint to receive price information from sap

### PR DESCRIPTION
**Pull request by Vishal14583**
- UnitOfSale event endpoint
- Mapping price information
- Append price information to  ENC event data
- Unit tests
- Few naming conventions uplifting
- Functional tests are in progress

### CODE REVIEW

File: src/UKHO.ERPFacade.Common/IO/Azure/AzureBlobEventWriter.cs [hunk 1]
```diff
@@ -1,5 +1,7 @@
 ﻿using Azure.Storage.Blobs;
+using Azure.Storage.Blobs.Models;
 using Microsoft.Extensions.Options;
+using Newtonsoft.Json.Linq;
```
Comment (Shirin14926) [inline src/UKHO.ERPFacade.Common/IO/Azure/AzureBlobEventWriter.cs:4]: Remove unused namespace.

**Review by Shirin14926 — COMMENTED** [review]
Few minor comments please

File: src/UKHO.ERPFacade.API/Controllers/ErpFacadeController.cs [hunk 2]
```diff
@@ -12,43 +17,83 @@ public class ErpFacadeController : BaseController<ErpFacadeController>
...
+        private const string CorrIdKey = "corrid";
+        private const string RequestFormat = "json";
+        private const int EventSizeLimit = 1000000;
 
         public ErpFacadeController(IHttpContextAccessor contextAccessor,
                                    ILogger<ErpFacadeController> logger,
                                    IAzureTableReaderWriter azureTableReaderWriter,
-                                   IAzureBlobEventWriter azureBlobEventWriter)
+                                   IAzureBlobEventWriter azureBlobEventWriter,
+                                   IErpFacadeService ErpFacadeService,
```
[the remark this answers is not in the document]
Comment (Vishal14583) [reply src/UKHO.ERPFacade.API/Controllers/ErpFacadeController.cs]: Updated

File: src/UKHO.ERPFacade.API/Helpers/IScenarioBuilder.cs [hunk 3]
```diff
@@ -4,6 +4,11 @@ namespace UKHO.ERPFacade.API.Helpers
 {
     public interface IScenarioBuilder
     {
-        List<Scenario> BuildScenarios(EESEvent eventData);
+        List<Scenario> BuildScenarios(EncEventPayload eventData);
+        //List<Scenario> BuildScenariosPayment(PriceInformationEvent priceInfoEvent);
```
[the remark this answers is not in the document]
Comment (Vishal14583) [reply src/UKHO.ERPFacade.API/Helpers/IScenarioBuilder.cs]: Removed

File: src/UKHO.ERPFacade.Common/IO/Azure/AzureBlobEventWriter.cs (hunk 1 above)
Comment (Vishal14583) [reply src/UKHO.ERPFacade.Common/IO/Azure/AzureBlobEventWriter.cs]: Removed

**Review by jrippington — APPROVED** [review]
Looks good to me

**Review by martyn-fewtrell — APPROVED** [review]
Looks good

**Pull request merged**